galasa-parent/dev.galasa.framework.api.cps/src/main/java/dev/galasa/framework/api/cps/internal/common/Namespace.java

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright contributors to the Galasa project
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package dev.galasa.framework.api.cps.internal.common;
+
+import java.util.HashSet;
+import java.util.Set;
+
+public class Namespace {
+
+   
+    private String name;
+    private String propertiesUrl;
+    private NamespaceType type;
+
+    public Namespace(String namespace){
+        this.name = namespace;
+        if (isSecureNamespace()){
+            this.type= NamespaceType.SECURE;
+        }else{
+            this.type=NamespaceType.NORMAL;
+        }
+    }
+
+    public Namespace(String namespace, String propertiesUrl){
+        this(namespace);
+        this.propertiesUrl = propertiesUrl+this.name+"/properties";
+    }
+
+
+    private static final String REDACTED_PROPERTY_VALUE = "********";
+
+    private static final Set<String> hiddenNamespaces = new HashSet<>();
+    static {
+        hiddenNamespaces.add("dss");
+    }
+
+    /**
+     * Some namespaces are able to be set, but cannot be queried.
+     *
+     * When they are queried, the values are redacted
+     */
+    private static final Set<String> secureNamespaces = new HashSet<>();
+    static {
+        secureNamespaces.add("secure");
+    }
```

Comment:
Really minor but I'd put constant declarations above methods and constructors. Also, not sure if it makes sense for each Namespace object to contain sets of hidden and secure namespaces - they would surely be stored elsewhere. Might become a nightmare to maintain if more hidden/secure namespaces are created!